### PR TITLE
C3: improve nuxt deployment

### PR DESCRIPTION
**Pull request by dario-piotrowicz**
improve nuxt deployment script so that it ships full stack applications (instead of server-side generated ones)

also write a .node-version file with the node version (so that it can properly working with the Pages CI)

Fixes # [insert GH or internal issue number(s)].

**What this PR solves / how to test:**

**Associated docs issue(s)/PR(s):**

- [insert associated docs issue(s)/PR(s)]

**Author has included the following, where applicable:**

- [ ] Tests
- [x] Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))

**Reviewer is to perform the following, as applicable:**

- Checked for inclusion of relevant tests
- Checked for inclusion of a relevant changeset
- Checked for creation of associated docs updates
- Manually pulled down the changes and spot-tested

### CODE REVIEW

File: packages/create-cloudflare/src/frameworks/nuxt/index.ts
Comment (dario-piotrowicz) [inline packages/create-cloudflare/src/frameworks/nuxt/index.ts:25]: I thought of separating the node version thing so that if someone were to use C3 to create a nuxt app but link it to the Pages CI then they would get the node version set correctly (instead of either having to set it manually or having to add it to their build command)

Comment (dario-piotrowicz) [reply packages/create-cloudflare/src/frameworks/nuxt/index.ts:25]: PS: 17 is also unsupported (https://endoflife.date/nodejs) so maybe we should just do 18? 🤷 

(also the build image v2 uses 18 by default I believe? so maybe there is no need to actually do anything? @GregBrimble)

**Review by jculvey — APPROVED** [review]

File: packages/create-cloudflare/src/frameworks/nuxt/index.ts
Comment (RamIdeas) [inline packages/create-cloudflare/src/frameworks/nuxt/index.ts:35]: Should the `npm run build` command ever be run without the `NITRO_PRESET=cloudflare-pages` env var? Should we move the env var into the build command?

Comment (dario-piotrowicz) [reply packages/create-cloudflare/src/frameworks/nuxt/index.ts:35]: mh.... yes that's a totally valid point... 🤔 

You shouldn't run `npm run build` without the env var
but I don't think there is a huge benefit in that since:
 - if you deploy via `pages:deploy` you do get the env var
 - `pages:dev` doesn't use `npm run build` so for local development there is no difference
 - if you want to deploy via the Pages CI you get the env var from the environment there already
 
 One issue here is that `pagesScripts` are used to add/override scripts in package.json:
https://github.com/cloudflare/workers-sdk/blob/113fb7fe86cde0c43a7efa7fc2e58ec8c55d8267/packages/create-cloudflare/src/pages.ts#L117
so I would either have to:
- refactor the `packageScripts` logic to somehow allow modification of scripts instead of overriding them (in this case prepending) 
- override the `build` script generated by the Nuxt CLI, which seems less robust than the current solution

🤔 
....

Actually the more I think of this the more refactoring the `packageScripts` seems like the most proper solution to me... what do you think?

@jculvey do you have an opinion on this? (and if you say we should update the logic, any preference on the api?)

Comment (jculvey) [reply packages/create-cloudflare/src/frameworks/nuxt/index.ts:35]: I'm a little hesitant to refactor because I can't think of many more use cases in c3 right now that would benefit from modifying commands, so I'm leaning towards just keeping `NITRO_PRESET` in the deploy command.

If we were to refactor it I think the API of `packageScripts` would change to take a function instead of a string, where the function receives the existing command. Something like: 

```
	packageScripts: {
		"pages:build": (cmd) => `NITRO_PRESET ${cmd}`,
    ...
  }
```

Without another use case though it's hard to tell if that enhancement would be enough to accommodate any future use cases we might encounter.

Comment (dario-piotrowicz) [reply packages/create-cloudflare/src/frameworks/nuxt/index.ts:35]: yeah that improvement looks like a good idea 🙂

anyways as I mentioned it might be not too important right now....

maybe we could go with the current implementation and if later people complain about it we just chance course?

I totally don't mind adding the above refactoring so that we still allow `packageScripts` to be strings but optionally if one is a function then we run it and use its output, it actually doesn't sound too difficult/problematic anyways...

I don't mind, whatever you guys prefer 🙂 (please make the final call 😁)

Comment (jculvey) [reply packages/create-cloudflare/src/frameworks/nuxt/index.ts:35]: `I totally don't mind adding the above refactoring so that we still allow packageScripts to be strings but optionally if one is a function then we run it and use its output, it actually doesn't sound too difficult/problematic anyways...`

Go for it!

Comment (dario-piotrowicz) [reply packages/create-cloudflare/src/frameworks/nuxt/index.ts:35]: Cool 🙂 , I've converted the PR to draft just to make sure we don't merge it until I implement that 🙂

Comment (dario-piotrowicz) [reply packages/create-cloudflare/src/frameworks/nuxt/index.ts]: Updated, please have a look @jculvey 🙏

**Review by jculvey — APPROVED** [review]
Tests might have to change when the interactive e2e tests PR lands. I guess it's a race 🐎

**Review by jculvey — APPROVED** [review]

**Pull request merged**
